Assets/BossRoom/Scripts/Client/Game/State/ClientCharSelectState.cs

```diff
@@ -171,6 +170,19 @@ private void UpdatePlayerCount()
         /// </summary>
         private void OnLobbyPlayerStateChanged(NetworkListEvent<CharSelectData.LobbyPlayerState> changeEvent)
         {
+            // ignore state changes for the local player unless the change event is a locked in event, or when seat
+            // has been reset to -1 (both server-authoritative)
+            if (changeEvent.Value.ClientId == NetworkManager.Singleton.LocalClientId)
+            {
+                var isLockedInEvent = (changeEvent.Value.SeatState == CharSelectData.SeatState.LockedIn && !m_HasLocalPlayerLockedIn) ||
```

Comment:
why do we use local values for locked in? With server authority, we should only take the server value as the authoritative value and never rely on local values

Reply:
So on OnPlayerClickedSeat(..), that's where the local player makes their anticipated selection, but also sends a ServerRpc for that change to be applied by the server to the NetworkList, which is then replicated to other clients.

Since the local player will still receive that same change event, I need to identify if it's just a local character selection change which can be ignored. What shouldn't be ignored is seat status changes.

NetworkListEvent doesn't provide that granularity on exactly what value was modified, so I'd need to compare it against _something_ locally, no?

I could also be comparing it against the current seat from `m_PlayerSeats`, since that is what is actually modified before the ServerRpc is sent. But that's still also something stored locally.

Reply:
discussed IRL, should have previous value coming from SDK, need following up with Fatih